sktime/forecasting/arch/_uarch.py

```diff
@@ -0,0 +1,362 @@
+        '1945-01-01'.
+    align : str, optional
+        Either 'origin' or 'target'. When set of 'origin', the t-th row of forecast
+        contains the forecasts for t+1, t+2, ..., t+h. When set to 'target', the
+        t-th row contains the 1-step ahead forecast from time t-1, the 2 step from
+        time t-2, ..., and the h-step from time t-h. 'target' simplified computing
+        forecast errors since the realization and h-step forecast are aligned.
+    method : {'analytic', 'simulation', 'bootstrap'}
+        Method to use when producing the forecast. The default is analytic.
+        The method only affects the variance forecast generation. Not all volatility
+        models support all methods.
+        In particular, volatility models that do not evolve in squares such as
+        EGARCH or TARCH
+        do not support the 'analytic' method for horizons > 1.
+    simulations : int
+        Number of simulations to run when computing the forecast using either
+        simulation or bootstrap.
+    rng : callable, optional
+        Custom random number generator to use in simulation-based forecasts.
+        Must produce random samples using the syntax rng(size) where size the
+        2-element tuple (simulations, horizon).
+    random_state : RandomState, optional
+        NumPy RandomState instance to use when method is 'bootstrap'
+    reindex : bool, optional
+        Whether to reindex the forecasts to have the same dimension as the series
+        being forecast. Prior to 4.18 this was the default. As of 4.19 this is now
+        optional. If not provided, a warning is raised about the future change in
+        the default which will occur after September 2021.
+
+    See Also
+    --------
+    StatsForecastARCH
+    StatsForecastGARCH
+
+    References
+    ----------
+    .. [1] Jason Brownlee. How to Model Volatility with ARCH and GARCH for Time Series
+       Forecasting.
+       https://machinelearningmastery.com/develop-arch-and-garch-models-for-time-series-
+       forecasting-in-python/
+    .. [2] GitHub repository of arch package (soft dependency).
+       https://github.com/bashtage/arch
+    .. [3] Documentation of arch package (soft dependency). Forecasting Volatility with
+       ARCH and it's variants.
+       https://arch.readthedocs.io/en/latest/univariate/introduction.html
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.garch import ARCH
+    >>> y = load_airline()
+    >>> forecaster = ARCH()
+    >>> forecaster.fit(y)
+    ARCH(...)
+    >>> y_pred = forecaster.predict(fh=1)
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": False,
+        "python_dependencies": "arch",
+        "ignores-exogeneous-X": True,
```

Comment:
Is this correctly specified? You are passing X to arch_model, which should use that as well in specific cases, e.g. ARX.  Will you modify it to be set conditionally based on mean argument?

Reply:
Okay, will check it.

Reply:
In that case, setting this tag as True gives a false impression to users that C is always ignored. I'll suggest to change it.

Reply:
I can suggest two options: 

1. You detect mean argument in init method and based on that set the tag.
2. You create separate classes for different models and set appropriate static tags.

I don't know what's better in an objective way.

Reply:
Okay, then I shall confirm it again and do the necessary things.